pyquil/paulis.py

```diff
@@ -94,13 +94,14 @@ def __len__(self):
     def get_qubits(self):
         """Gets all the qubits that this PauliTerm operates on.
         """
-        return list(self._ops.keys())
+        # sort the keys to get a deterministic iteration order over qubits
+        return sorted(self._ops.keys())
 
     def __getitem__(self, i):
         return self._ops.get(i, "I")
 
     def __iter__(self):
-        for i in self._ops.keys():
+        for i in self.get_qubits():
```

Comment:
@jotterbach there are multiple places where `sorted(self._ops.keys())` are still used that could be replaced with THIS.